File: designpatterns/src/com/renho/behavior/mediator/ConcreteMediatorA.java
package com.renho.behavior.mediator;

public class ConcreteMediatorA extends Mediator {

	private ConcreteColleagueA colleagueA;
	private ConcreteColleagueB colleagueB;
	
	@Override
	public void send(String message, Colleague colleague) {
		if(colleagueA == colleague) {
			colleagueB.notify(message);
		}else {
			colleagueA.notify(message);
		}
	}

	public ConcreteColleagueA getColleagueA() {
		return colleagueA;
	}

	public void setColleagueA(ConcreteColleagueA colleagueA) {
		this.colleagueA = colleagueA;
	}

	public ConcreteColleagueB getColleagueB() {
		return colleagueB;
	}

	public void setColleagueB(ConcreteColleagueB colleagueB) {
		this.colleagueB = colleagueB;
	}
}

File: ideaworkspace/designpatterns/src/main/scala/com/renho/behavior/mediator/ConcreteMediatorA.scala
package com.renho.behavior.mediator

/**
  * Created by renho on 2016/7/11.
  */
class ConcreteMediatorA extends Mediator {
	var cca: ConcreteColleagueA = _
	var ccb: ConcreteColleagueB = _
	override def send(message: String, colleague: Colleague): Unit = {
		colleague match {
			case ConcreteColleagueA(mediator) => ccb.notify(message)
			case ConcreteColleagueB(mediator) => cca.notify(message)
			case _ => println("no...")
		}
	}
}
